dynamiq/evaluations/metrics/context_precision.py

```diff
@@ -0,0 +1,287 @@
+import logging
+
+from pydantic import BaseModel, PrivateAttr, field_validator, model_validator
+
+from dynamiq.components.evaluators.llm_evaluator import LLMEvaluator
+from dynamiq.nodes.llms import BaseLLM
+
+# Configure logging
+logger = logging.getLogger(__name__)
+
+
+class ContextPrecisionInput(BaseModel):
+    """
+    Input model for context precision evaluation.
+
+    Attributes:
+        questions (List[str]): List of questions.
+        answers (List[str]): List of corresponding answers.
+        contexts_list (List[List[str]]): List of contexts for each question.
+        verbose (bool): Flag to enable verbose logging.
+    """
+
+    questions: list[str]
+    answers: list[str]
+    contexts_list: list[list[str]]
+    verbose: bool = False
+
+    @model_validator(mode="after")
+    def check_equal_length(self):
+        if not (len(self.questions) == len(self.answers) == len(self.contexts_list)):
+            raise ValueError("Questions, answers, and contexts_list must have the same length.")
+        return self
+
+
+class ContextPrecisionOutput(BaseModel):
+    """
+    Output model for context precision evaluation.
+
+    Attributes:
+        final_scores (List[float]): List of context precision scores.
+    """
+
+    final_scores: list[float]
+
+
+class VerdictResult(BaseModel):
+    """
+    Model for the verdict result from the evaluator.
+
+    Attributes:
+        verdict (int): 1 if the context was useful, 0 otherwise.
+        reason (str): Reason for the verdict.
+    """
+
+    verdict: int
+    reason: str
+
+    @field_validator("verdict")
+    @classmethod
+    def validate_verdict(cls, v):
+        if v not in (0, 1):
+            raise ValueError("Verdict must be either 0 or 1.")
+        return v
+
+
+class ContextPrecisionEvaluator(BaseModel):
+    """
+    Evaluator class for context precision metric.
+
+    Attributes:
+        llm (BaseLLM): The language model to use for evaluation.
+    """
+
+    llm: BaseLLM
+
+    # Private attribute (not part of the Pydantic model fields)
+    _context_precision_evaluator: LLMEvaluator = PrivateAttr()
+
+    class Config:
+        arbitrary_types_allowed = True  # Allow arbitrary types
+
```

Comment:
I think we should add type for each evaluator (probably with some BaseEvaluator to limit duplication) in similar way as we define for Nodes:
https://github.com/dynamiq-ai/dynamiq/blob/main/dynamiq/nodes/node.py#L231

```
    @computed_field
    @cached_property
    def type(self) -> str:
        return f"{self.__module__.rsplit('.', 1)[0]}.{self.__class__.__name__}"
```
As well need to modify `__init__.py`
```
from .context_precision import ContextPrecisionEvaluator
```

As result we could get proper class by type:
```
from dynamiq.loaders.yaml import WorkflowYAMLLoader

metric_evaluator_cls = WorkflowYAMLLoader.get_entity_by_type("dynamiq.evaluations.metrics.ContextPrecisionEvaluator")
```